tsl/src/remote/tuplefactory.c

```diff
@@ -90,22 +90,25 @@ conversion_error_callback(void *arg)
 	{
 		/* error occurred in a scan against a foreign join */
 		ScanState *ss = errpos->ss;
-		ForeignScan *fsplan;
+		List *tlist;
 		EState *estate = ss->ps.state;
 		TargetEntry *tle;
 
 		if (IsA(ss->ps.plan, ForeignScan))
-			fsplan = (ForeignScan *) ss->ps.plan;
+		{
+			ForeignScan *fsplan = (ForeignScan *) ss->ps.plan;
```

Comment:
This error reporting code used to segfault, I fixed it and added a test to dist_hypertable.